xarray/core/computation.py

```diff
@@ -1306,3 +1307,86 @@ def where(cond, x, y):
         dataset_join="exact",
         dask="allowed",
     )
+
+
+def _calc_idxminmax(
+    *,
+    array,
+    func: Callable,
+    dim: Optional[Hashable],
+    skipna: Optional[bool],
+    promote: Optional[bool],
+    keep_attrs: Optional[bool],
+    **kwargs: Any,
+):
+    """Apply common operations for idxmin and idxmax."""
+    # This function doesn't make sense for scalars so don't try
+    if not array.ndim:
+        ValueError("This function does not apply for scalars")
+
+    if dim is not None:
+        pass  # Use the dim if available
+    elif array.ndim == 1:
+        # it is okay to guess the dim if there is only 1
+        dim = array.dims[0]
+    else:
+        # The dim is not specified and ambiguous.  Don't guess.
+        raise ValueError("Must supply 'dim' argument for multidimensional arrays")
+
+    if dim in array.coords:
+        pass  # This is okay
+    else:
+        raise KeyError(f'Dimension "{dim}" does not have coordinates')
+
+    # These are dtypes with NaN values argmin and argmax can handle
+    na_dtypes = "cf0"
+
+    if skipna or (skipna is None and array.dtype.kind in na_dtypes):
```

Comment:
I think the right check here is probably:
```suggestion
    if skipna or (skipna is None and issubclass(array.dtype.type, np.inexact):
```
This will handle all complex/float types: https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.scalars.html#arrays-scalars-built-in

Reply:
This is supposed to very specifically test the types that have NA types xarray can handle.  This particular test is used throughout xarray.

Reply:
Looks good!